src/sentry/services/hybrid_cloud/app/impl.py

```diff
@@ -203,14 +203,15 @@ def apply_filters(
                 # the installation by token id in SentryAppInstallation, we also search
                 # SentryAppInstallationToken by token id, then fetch  the linked installation.
                 # Internal Integrations follow this pattern because they can have multiple tokens.
-                from django.db.models import Q, Subquery
 
-                subquery = SentryAppInstallationToken.objects.filter(
+                # Decompose this query instead of using a subquery for performance.
+                install_token_list = SentryAppInstallationToken.objects.filter(
                     api_token_id=filters["api_installation_token_id"],
-                ).values("sentry_app_installation_id")
+                ).values_list("sentry_app_installation_id", flat=True)
+
                 query = query.filter(
                     Q(api_token_id=filters["api_installation_token_id"])
-                    | Q(id__in=Subquery(subquery))
+                    | Q(id__in=list(install_token_list))
```

Comment:
I added a `list()` cast to force django to evaluate the nested query. Without the list() django is 'helpful' and inserts a subquery for you.